Lean/README.md

```diff
@@ -1,12 +1,29 @@
 # Setup
 
-The only assumption on your installation is that the Lean binaries can be found in the folder
-`lean-3.4.1-linux/bin/`. This should also work for newer versions of lean.
-To change the path for the binaries, you can modify the variable `lean_bin`
-in `grader.py`.
+## First Time
+Run `preparejudge.sh`. The script installs a grader compatible with the lean version specified in `variables/lean_version` to the path specified in `variables/grader_folder/${lean_version}`.
 
 Before running the first time, you need to copy `config.tmp` to `config`.
 Then insert the correct URL for the frontend server and your access token.
 
-Now, the backend can simply be administered via
-`./judge start`, `./judge status`, and `./judge stop`.
\ No newline at end of file
+## Start the judge
+The backend can simply be administered via
+`./judge start`, `./judge status`, and `./judge stop`.
+
+It expects the following structure of the submitted files:
+
++-- _defs.lean_
++-- _submission.lean_
++-- _check.lean_
+
+The poller checks all lemmas/theorems in `check.lean`. 
+
+## Tests
+Tests can be executed using `tests_run.sh` and `tests.py`.
+
+## TODOs
+- firejail needs to be fixed in `grader_run.sh`
+- Mathlib is not used by the grader.
```

Comment:
Should we allow to include mathlib? It contains most of Lean's mathematical library and many useful tactics.